lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/HandlerContainer.java

```diff
@@ -79,14 +70,7 @@ public AutoCloseable registerHandler(final RemoteIdentifier sourceIdentifier,
   public AutoCloseable registerHandler(
       final Class<? extends T> messageType,
       final EventHandler<RemoteMessage<? extends T>> theHandler) {
-
-    final EventHandler<RemoteMessage<? extends T>> prevHandler =
-        this.msgTypeToHandlerMap.put(messageType, theHandler);
```

Comment:
By the way, may I change them with `putIfAbsent` and `get` instead in this context? I think that would be better since concurrent hash map becomes append-only.